File: Presentation/Client/src/app/models/CartItem.ts


export class CartItem {
  ProductId: number;
  Name: string;
  UnitPrice: number;
  Quantity: number;
  Total: number;

  constructor( productId: number,  name: string,  unitPrice: number,  quantity: number) {
    this.ProductId = productId;
    this.Name = name;
    this.UnitPrice = unitPrice;
    this.Quantity = quantity;
    this.Total = quantity * unitPrice;
  }
}

File: DomainModels/Entities/CartItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DomainModels.Entities
{
	public class CartItem
	{
		[Key]
		public int CartItemId { get; set; }
		public int ProductId { get; set; }

		[Column(TypeName = "decimal(18,2)")]
		public decimal UnitPrice { get; set; }
		public int Quantity { get; set; }
		public int Total { get; set; }

		[ForeignKey("Cart")]
		public int CartId { get; set; }

		public virtual Cart Cart { get; set; }
		public virtual Product Product { get; set; }
	}
}